src/framework/constraints.ml

```diff
@@ -554,10 +554,16 @@ struct
       (* TODO: don't repeat for all paths that spawn same *)
       let ds = S.threadenter ctx lval f args in
       List.iter (fun d ->
-          let c = S.context d in
           spawns := (lval, f, args, d) :: !spawns;
-          if not full_context then sidel (FunctionEntry f, c) d;
-          ignore (getl (Function f, c))
+          match Cilfacade.getdec f with
+          | fd ->
+            let c = S.context d in
+            if not full_context then sidel (FunctionEntry fd, c) d;
+            ignore (getl (Function fd, c))
+          | exception Not_found ->
+            (* unknown function *)
+            M.warn_each ("Created a thread from unknown function " ^ f.vname)
```

Comment:
Maybe we should phrase this differently because base might decide to not create a thread here depending on the type.

Reply:
If base decides to not create it based on the type, this `ctx.spawn` is never called anyway.